hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteBufferedP.java

```diff
@@ -52,7 +53,11 @@
 
     @Override
     public void process(int ordinal, @Nonnull Inbox inbox) {
-        inbox.drain((T t) -> addToBuffer.accept(buffer, t));
+        for (Object o; (o = inbox.poll()) != null; ) {
+            if (!(o instanceof Punctuation)) {
+                addToBuffer.accept(buffer, (T) o);
+            }
+        }
```

Comment:
I also prefer the `drain(Consumer)` call. You're repeating the same for loop which is inside that call already.